Base/res/ladybird/about-pages/settings/privacy.js

```diff
@@ -1,15 +1,125 @@
+const clearBrowsingData = document.querySelector("#clear-browsing-data");
+const clearBrowsingDataCachedFiles = document.querySelector("#clear-browsing-data-cached-files");
+const clearBrowsingDataCachedFilesSize = document.querySelector("#clear-browsing-data-cached-files-size");
+const clearBrowsingDataClose = document.querySelector("#clear-browsing-data-close");
+const clearBrowsingDataDialog = document.querySelector("#clear-browsing-data-dialog");
+const clearBrowsingDataRemoveData = document.querySelector("#clear-browsing-data-remove-data");
+const clearBrowsingDataSiteData = document.querySelector("#clear-browsing-data-site-data");
+const clearBrowsingDataSiteDataSize = document.querySelector("#clear-browsing-data-site-data-size");
+const clearBrowsingDataTimeRange = document.querySelector("#clear-browsing-data-time-range");
+const clearBrowsingDataTotalSize = document.querySelector("#clear-browsing-data-total-size");
 const globalPrivacyControlToggle = document.querySelector("#global-privacy-control-toggle");
 
+const BYTE_UNITS = ["byte", "kilobyte", "megabyte", "gigabyte", "terabyte"];
+
+const BYTE_FORMATTERS = {
+    byte: undefined,
+    kilobyte: undefined,
+    megabyte: undefined,
+    gigabyte: undefined,
+    terabyte: undefined,
+};
+
+function formatBytes(bytes) {
+    let index = 0;
+    while (bytes >= 1024 && index < BYTE_UNITS.length - 1) {
+        bytes /= 1024;
+        ++index;
+    }
+
+    const unit = BYTE_UNITS[index];
+
+    if (!BYTE_FORMATTERS[unit]) {
+        BYTE_FORMATTERS[unit] = new Intl.NumberFormat([], {
+            style: "unit",
+            unit: unit,
+            unitDisplay: unit === "byte" ? "long" : "short",
+            maximumFractionDigits: 1,
+        });
+    }
```

Comment:
this is suspicious: it seems it should be global, but then you would not be able to make it depend on `unit`/`index`;

maybe something like:

```ts
    const createFormat = (unit) => new Intl.NumberFormat([], {
        style: "unit",
        unit: unit,
        unitDisplay: unit === "byte" ? "long" : "short",
        maximumFractionDigits: 1,
    });

    const BYTE_FORMATTERS = Object.fromEntries(
        ["byte", "kilobyte", "megabyte", "gigabyte", "terabyte"]
            .map(unit => [unit, createFormat(unit)]
    );
```